compiler/backend/presLangScript.sml

```diff
@@ -234,381 +234,341 @@ val exh_to_pres_exp_def = tDefine"exh_to_pres_exp"`
+  (op_to_structured (Ast_op Aw8length) = Item NONE "Aw8length" [])
   /\
-  (op_to_json (Ast_op Aw8update) = new_obj "Aw8update" [])
+  (op_to_structured (Ast_op Aw8update) = Item NONE "Aw8update" [])
   /\
-  (op_to_json (Ast_op (WordFromInt word_size)) = new_obj "WordFromInt" [
-    ("word_size", word_size_to_json word_size)
-  ])
+  (op_to_structured (Ast_op (WordFromInt word_size)) = Item NONE "WordFromInt" [
+    word_size_to_structured word_size ])
   /\
-  (op_to_json (Ast_op (WordToInt word_size)) = new_obj "WordToInt" [
-    ("word_size", word_size_to_json word_size)
-  ])
+  (op_to_structured (Ast_op (WordToInt word_size)) = Item NONE "WordToInt" [
+    word_size_to_structured word_size ])
   /\
-  (op_to_json (Ast_op Ord) = new_obj "Ord" [])
+  (op_to_structured (Ast_op Ord) = Item NONE "Ord" [])
   /\
-  (op_to_json (Ast_op Chr) = new_obj "Chr" [])
+  (op_to_structured (Ast_op Chr) = Item NONE "Chr" [])
   /\
-  (op_to_json (Ast_op (Chopb opb)) = new_obj "Chopb" [("opb", opb_to_json opb)])
+  (op_to_structured (Ast_op (Chopb opb)) = Item NONE "Chopb" [opb_to_structured opb])
   /\
-  (op_to_json (Ast_op Implode) = new_obj "Implode" [])
+  (op_to_structured (Ast_op Implode) = Item NONE "Implode" [])
   /\
-  (op_to_json (Ast_op Strsub) = new_obj "Strsub" [])
+  (op_to_structured (Ast_op Strsub) = Item NONE "Strsub" [])
   /\
-  (op_to_json (Ast_op Strlen) = new_obj "Strlen" [])
+  (op_to_structured (Ast_op Strlen) = Item NONE "Strlen" [])
   /\
-  (op_to_json (Ast_op VfromList) = new_obj "VfromList" [])
+  (op_to_structured (Ast_op VfromList) = Item NONE "VfromList" [])
   /\
-  (op_to_json (Ast_op Vsub) = new_obj "Vsub" [])
+  (op_to_structured (Ast_op Vsub) = Item NONE "Vsub" [])
   /\
-  (op_to_json (Ast_op Vlength) = new_obj "Vlength" [])
+  (op_to_structured (Ast_op Vlength) = Item NONE "Vlength" [])
   /\
-  (op_to_json (Ast_op Aalloc) = new_obj "Aalloc" [])
+  (op_to_structured (Ast_op Aalloc) = Item NONE "Aalloc" [])
   /\
-  (op_to_json (Ast_op Asub) = new_obj "Asub" [])
+  (op_to_structured (Ast_op Asub) = Item NONE "Asub" [])
   /\
-  (op_to_json (Ast_op Alength) = new_obj "Alength" [])
+  (op_to_structured (Ast_op Alength) = Item NONE "Alength" [])
   /\
-  (op_to_json (Ast_op Aupdate) = new_obj "Aupdate" [])
+  (op_to_structured (Ast_op Aupdate) = Item NONE "Aupdate" [])
   /\
-  (op_to_json (Ast_op (FFI str)) = new_obj "FFI" [("str", String str)])
+  (op_to_structured (Ast_op (FFI str)) = Item NONE "FFI" [string_to_structured str])
   /\
-  (op_to_json _ = new_obj "Unknown" [])`;
+  (op_to_structured _ = Item NONE "Unknown" [])`;
 
-val lop_to_json_def = Define`
-  (lop_to_json ast$And = String "And")
+val lop_to_structured_def = Define`
+  (lop_to_structured ast$And = string_to_structured "And")
   /\
-  (lop_to_json Or = String "Or")
+  (lop_to_structured Or = string_to_structured "Or")
   /\
-  (lop_to_json _ = String "Unknown")`
+  (lop_to_structured _ = string_to_structured "Unknown")`
```

Comment:
Is this case necessary? The `lop` datatype in `astScript` only defines `And` and `Or`, and I suspect there won't be any changes to that for the foreseeable future.

Also, nit: add a semicolon after `.